staging/src/k8s.io/client-go/rest/transport.go

```diff
@@ -58,25 +58,20 @@ func HTTPWrappersForConfig(config *Config, rt http.RoundTripper) (http.RoundTrip
 
 // TransportConfig converts a client config to an appropriate transport config.
 func (c *Config) TransportConfig() (*transport.Config, error) {
-	wt := c.WrapTransport
-	if c.AuthProvider != nil {
-		provider, err := GetAuthProvider(c.Host, c.AuthProvider, c.AuthConfigPersister)
-		if err != nil {
-			return nil, err
-		}
-		if wt != nil {
-			previousWT := wt
-			wt = func(rt http.RoundTripper) http.RoundTripper {
-				return provider.WrapTransport(previousWT(rt))
+	if c.BearerTokenSource == nil {
+		if c.AuthProvider != nil {
+			provider, err := GetAuthProvider(c.Host, c.AuthProvider, c.AuthConfigPersister)
+			if err != nil {
+				return nil, err
 			}
-		} else {
-			wt = provider.WrapTransport
+			c.BearerTokenSource = provider.BearerTokenSource()
```

Comment:
Users should be reusing the transport.Config instead of calling it multiple times anyway. It's just unusual to see something modify the receiver like this.